src/pandas_profiling/expectations_report.py

```diff
@@ -0,0 +1,91 @@
+from pandas_profiling.model import expectation_algorithms
+from pandas_profiling.model.handler import Handler
+from pandas_profiling.model.typeset import (
+    URL,
+    Boolean,
+    Categorical,
+    DateTime,
+    File,
+    Image,
+    Numeric,
+    Path,
+    ProfilingTypeSet,
+    Unsupported,
+)
+from pandas_profiling.utils.dataframe import slugify
+
+
+# Default handler
+class ExpectationHandler(Handler):
+    def __init__(self, typeset, *args, **kwargs):
+        mapping = {
+            Unsupported: [expectation_algorithms.generic_expectations],
+            Categorical: [expectation_algorithms.categorical_expectations],
+            Numeric: [expectation_algorithms.numeric_expectations],
+            URL: [expectation_algorithms.url_expectations],
+            File: [expectation_algorithms.file_expectations],
+            Path: [expectation_algorithms.path_expectations],
+            Boolean: [expectation_algorithms.boolean_expectations],
+            DateTime: [expectation_algorithms.datetime_expectations],
+            Image: [expectation_algorithms.image_expectations],
+        }
+        super().__init__(mapping, typeset, *args, **kwargs)
+
+
+class ExpectationsReport:
+    def to_expectations_suite(self, suite_name=None, handler=None):
+        try:
+            import great_expectations as ge
+        except ImportError:
+            raise ImportError(
+                "Please install great expectations before using the expectation functionality"
+            )
+
+        # Use report title if suite is empty
+        if suite_name is None:
+            suite_name = slugify(self.title)
+
+        # Use the default handler if none
+        if handler is None:
+            handler = ExpectationHandler(ProfilingTypeSet())
+
+        # Obtain the ge context and create the expectation suite
+        context = ge.data_context.DataContext()

```

Comment:
One suggestion from Sam - we've recently been improving our in-memory data contexts, and we're starting to get users who are building data contexts in that way. If we allow users to pass a data context into the function (and only instantiate a data context if nothing is passed in), then it would also support those users.